exporter/factory.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/census-instrumentation/opencensus-service/processor"
+)
+
+// TraceDataExporter inherits functions from TraceDataProcessor, and additionally
+// adds some exporter-specific functions. This helps the collector core to
+// identify which TraceDataProcessors are Exporters and which are internal
+// processing components, so that better validation of pipelines can be done.
+type TraceDataExporter interface {
+	processor.TraceDataProcessor
+
+	// GetExportFormat gets the name of the format in which this exporter sends its data.
+	GetExportFormat() string
+}
+
+// TraceDataExporterFactory is an interface that builds a new TraceDataExporter based on
+// some viper.Viper configuration.
+type TraceDataExporterFactory interface {
+	// GetType gets the type of the TraceDataExporter created by this factory.
+	GetType() string
+	// NewFromViper takes a viper.Viper config and creates a new TraceDataExporter.
```

Comment:
What about having the two viper related methods that are config related in a separate interface? My reasoning is that we will need to create things from the config that are not exporters, processors, etc. That interface becomes what anything that is created from the config needs to implement.

Reply:
the downside is the interface to be returned... hum...

Reply:
Yea i prefer the strong typing, I think

Reply:
at minimum the interface could cover every type thata we can put and connect on the pipeline.

Reply:
`TraceDataExporter` is an extension of `TraceDataProcessor`, so `TraceDataProcessor` is basically what you are talking about I think. I think keeping the typing like this will allow us to use `TraceDataExporter`s as `TraceDataProcessor`s when its convenient, while still allowing us to know the functional difference that `TraceDataExporter`s are required as the end of pipeline.

Reply:
Let's go your way, it seems better at this stage.